sdk/eventhub/azure-eventhub/azure/eventhub/exceptions.py

```diff
@@ -36,7 +38,7 @@ def _error_handler(error):
     return errors.ErrorAction(retry=True)
 
 
-class EventHubError(Exception):
+class EventHubError(AzureError):
```

Comment:
As now the `EventHubError` inherits from the `AzureError`, I think we need to call the `super.__init__()` in the end of the constructor as following:

```
if details and isinstance(details, Exception):
    super(EventHubError, self).__init__(self.message, error=details)
else:
    super(EventHubError, self).__init__(self.message)
```

Reply:
Hey @yunhaoling ; thanks for the review; do you mean that I should replace the current `__init__()` with something like this? 
```
def __init__(self, message, details=None):
        self.error = None
        self.message = message
        self.details = details
        if details and isinstance(details, Exception):
            try:
                condition = details.condition.value.decode("UTF-8")
            except AttributeError:
                try:
                    condition = details.condition.decode("UTF-8")
                except AttributeError:
                    condition = None
            if condition:
                _, _, self.error = condition.partition(":")
                self.message += "\nError: {}".format(self.error)
            try:
                self._parse_error(details.description)
                for detail in self.details:
                    self.message += "\n{}".format(detail)
            except:  # pylint: disable=bare-except
                self.message += "\n{}".format(details)
        if details and isinstance(details, Exception):
             super(EventHubError, self).__init__(self.message, error=details)
        else:
             super(EventHubError, self).__init__(self.message)
```


or should I replace it with this?
```
def __init__(self, message, details=None):
        self.error = None
        self.message = message
        self.details = details
        if details and isinstance(details, Exception):
              super(EventHubError, self).__init__(self.message, error=details)
        else:
              super(EventHubError, self).__init__(self.message)
        
```

Or should I replace it with something like this? 

```
def __init__(self, message, details=None):
        self.error = None
        self.message = message
        self.details = details
        if details and isinstance(details, AzureError):
              super(EventHubError, self).__init__(self.message, error=details)
        else:
              super(EventHubError, self).__init__(self.message)
```

Reply:
Hey @yunhaoling , just wanted to ping this again. I'm not in a rush or anything, but also don't want this to get lost in a sea of notifs :)

Reply:
hey @bradleydamato , sorry for the late response.

I would expect it to be the first one, other approaches would break the message information completeness.

Reply:
@yunhaoling no problem on the late response, pushing the commit now.

Reply:
@yunhaoling unsure if the Azure Pipelines checks will auto-run again or not with my new commits. Is there a heuristic that would allow me to figure out whether they will run again or not? Either way, no rush on my end :).

Reply:
you did great work, thanks!
there are two minor indentation issue, I've fixed them.